training/analysis_functions.py

```diff
@@ -26,9 +27,42 @@ def unpack_column(
     :param print_info: print or not: first row and Len of that row
     :return: DataFrame with values from chosen column
     """
-    new_df = df[column_name].apply(pd.Series)
+    with warnings.catch_warnings():
+        warnings.simplefilter(action="ignore", category=FutureWarning)
+        new_df = df[column_name].apply(pd.Series)
+
     if print_info:
         print("unpack_column: First row:")
         print(new_df.iloc[0])
         print(f"Len = {len(new_df.iloc[0])}")
     return new_df
+
+
+def _create_null_dataframe(index: list[int], columns: list[str]):
+    """
+    Creates a dataframe which contains only null values
+
+    :param index: List of indices to use for resulting frame
+    :param columns: Column labels to use for resulting frame
+    :return: DataFrame with given labels and columns and null data
+    """
+    return pd.DataFrame(index=index, columns=columns)
+
+
+def unpack_column_with_null(df: pd.DataFrame, column_name: str) -> pd.DataFrame:
+    """
+    Unpacks chosen dictionary column even if there are with rows with only null value in that column and changes it to DataFrame.
```

Comment:
Hm... nie bardzo rozumiem to z nullami. Może warto byłoby tu zawrzeć jakiś przykład na dosłownie 2-3 wiersze?